.github/workflows/rust-tests.yml

```
@@ -8,6 +8,18 @@ on:
       - main
 
 jobs:
+  feature-checks:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v3
```

I think doing a recursive checkout will fix the feature checks issue.
```suggestion
      - uses: actions/checkout@v3
        with:
          submodules: recursive
```